components/hydrator/internal/tenantmapping/consumer_context_provider.go

```diff
@@ -96,12 +107,30 @@ func (c *consumerContextProvider) Match(_ context.Context, data oathkeeper.ReqDa
 }
 
 func (c *consumerContextProvider) getUserContextData(userContextHeader string) (*userContextData, error) {
-	clientID := gjson.Get(userContextHeader, c.consumerClaimsKeysConfig.ClientIDKey)
+	decodedUserContextHeader, err := url.QueryUnescape(userContextHeader)
+	if err != nil {
```

Comment:
There is a fair amount of code duplication.

The same can be achieved with

```go
if decodedUserContextHeader, err := url.QueryUnescape(userContextHeader); err == nil {
    userContextHeader = decodedUserContextHeader
}
```

And everything below stays the same.